ultralytics/solutions/serve.py

```diff
@@ -0,0 +1,29 @@
+import base64
+import io
+
+import litserve as ls
+import numpy as np
+from PIL import Image
+
+from ultralytics import YOLO
+
+
+class YOLOServe(ls.LitAPI):
+    def setup(self, device):
+        self.model = YOLO("yolo11n.pt")
+
+    def decode_request(self, request):
+        base64_str = request["input"]
+        img_data = base64.b64decode(base64_str)
+        image = Image.open(io.BytesIO(img_data))
+        image_array = np.array(image)
+        return image_array
+
+    def predict(self, x):
+        prediction = self.model(x)
+        return prediction[0].to_json()
```

Comment:
@ambitious-octopus with this change, [we can utilize auto batching support of the litserve](https://lightning.ai/docs/litserve/features/batching#enable-batching):

```suggestion
        return prediction
        
    def encode_response(self, results: ultralytics.engine.results.Results) -> UltralyticsResult:
        image_results = []
        for result in results
            image_results.append(result.to_json())
        return UltralyticsResult(results=results)
```

It would be great to define the request and response models for automated swagger API docs generation:
```python
from pydantic import BaseModel

class UltralyticsResult(BaseModel):
    status: str = "success"
    results: List[Dict]
```